cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,377 @@
+            )
+            if check_case_names_match(csv_case_name, case_name):
+                if not any(
+                    cluster.id == match.cluster.id
+                    for cluster, citation in matches
+                ):
+                    # Avoid duplicates
+                    matches.append(
+                        (match.cluster, citation.corrected_citation())
+                    )
+
+    return matches
+
+
+def update_matched_case_name(
+    matched_cluster: OpinionCluster, csv_case_name: str
+) -> bool:
+    """Update case name of matched cluster and related docket
+
+    :param matched_cluster: OpinionCluster object
+    :param csv_case_name: case name from csv row
+    :return: tuple with boolean values if cluster and related docket case name updated
+    """
+
+    if not matched_cluster.case_name or len(csv_case_name) < len(
+        matched_cluster.case_name
+    ):
+        # Save case name in cluster when we don't have it or when the case name in csv is smaller than the current case name
+        matched_cluster.case_name = csv_case_name
+        matched_cluster.save()
+        logger.info(f"Case name updated for cluster id: {matched_cluster.id}")
+        return True
+
+    logger.info(
+        f"Cluster id: {matched_cluster.id} already has the smallest case name."
+    )
+
+    return False
+
+
+def get_citation_filter(citation: FullCaseCitation) -> dict:
+    """Get citation as a dict to use it as a filter
+
+    2012-635 (La.App. 3 Cir. 12/5/12) also includes date_filed when it is parsed
+
+    :param citation:
+    :return: dict with volume, reporter and page
+    """
+    return {
+        "volume": citation.groups["volume"],
+        "reporter": citation.corrected_reporter(),
+        "page": citation.groups["page"],
+    }
+
+
+def process_csv(
+    filepath: str, delay: float, dry_run: bool, chunk_size: int
+) -> None:
+    """Process rows from csv file
+
+    :param filepath: path to csv file
+    :param delay: delay between saves in seconds
+    :param dry_run: flag to simulate update process
+    :param chunk_size: number of rows to read at a time
+    """
+
+    total_clusters_updated = 0
+    logger.info(f"Processing {filepath}")
+    for chunk in pd.read_csv(filepath, chunksize=chunk_size):
+        for row in chunk.dropna().itertuples():
+            (
+                index,
+                csv_case_name,
+                court,
+                date_str,
+                cite1,
+                cite2,
+                docket,
+                volume,
+            ) = row
+            valid_citations = make_citations(cite1, cite2)
+
+            if len(valid_citations) < 2:
+                # Skipping row without two citations
+                continue
+
+            if not NUMBER_PATTERN.match(
+                valid_citations[0].groups.get("volume")
+            ) or not NUMBER_PATTERN.match(
+                valid_citations[1].groups.get("volume")
+            ):
```

Comment:
I'm asking why not put this code in the parsing of valid citations.  instead of here.